pkg/identity/cache/local.go

```diff
@@ -62,7 +72,9 @@ func (l *localIdentityCache) getNextFreeNumericIdentity(idCandidate identity.Num
 	firstID := l.nextNumericIdentity
 	for {
 		idCandidate = l.nextNumericIdentity | l.scope
-		if _, taken := l.identitiesByID[idCandidate]; !taken {
+		_, taken := l.identitiesByID[idCandidate]
+		_, reserved := l.reservedIdentities[idCandidate]
```

Comment:
How does the callpath from ipcache determine that it is the legitimate one allocating the identity with this reserved numeric identity? I don't see any unreserve call or special path where that caller passes down the context to skip the reserve check.

Reply:
Oh, I misunderstood the intent of these changes. I wasn't familiar with this code, saw the changes to `idCandidate`, saw  this was the parameter to the function, and assumed that this path is about returning the parameter. I missed that the `idCandidate` variable is reused in this loop to iterate through the number space. Basically this code is implementing the protection of the reserved identity, not trying to return the parameter the caller gives.

Makes sense :+1: , I retract my concern here.